code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -593,6 +593,17 @@
 	if(candidates.len)
 		theghost = pick(candidates)
 		spawn_guardian(user, theghost.key)
+		for(var/mob/living/simple_animal/hostile/guardian/G in living_mob_list)
+			if (G.summoner == user)
+				var/time_limit = world.time + 300
+				var/guardianNewName = stripped_input(G.summoner, "You are the user of [G.name]. Would you like to name your guardian something else?", "Name Guardian", G.name)
```

Comment:
You've put no limit in this meaning that the name's limit is,by default,MAX_MESSAGE_LEN. You need to add an argument, which i suggest to be MAX_NAME_LEN	  
cause from what you can see here:
#define MAX_MESSAGE_LEN			1024
#define MAX_NAME_LEN			26
#define MAX_BROADCAST_LEN		512
having the name be of 1024 characters would be weird, but 26 chars look perfect